src/transport/MessageEndpoint.cpp

```diff
@@ -0,0 +1,303 @@
+#include <faabric/transport/MessageEndpoint.h>
```

Comment:
This has become the only file where we interact with ZeroMQ. After tedious debugging, I've wrapped all zeromq calls in `try/catch` blocks as otherwise exceptions were lost sometimes.

As a consequence, this file is now quite verbose, but we have hid the complexity from the others.